### PR TITLE
Organization Role

### PR DESCRIPTION
**Pull request by hindman-redhat**
Draft pull request

### CODE REVIEW

**Comment by willtome** [comment]
@hindman-redhat, currently the variables data structure would look like this
```
foreman_organizations: 
  - name: raleigh
    label: rdu
    server_url: url
    username: user
    password: pass
    validate_certs: no
    state: present
  - name: lanai 
    description: pacific datacenter 
    server_url: url
    username: user
    password: pass
    validate_certs: no
    state: present
  - name: default
    label: boring
    server_url: url
    username: user
    password: pass
    validate_certs: no
    state: present
```

Update so that the authentication vars, `server_url`, `username`, `password`, etc, so they are given once per role invocation with a list of the unique variables `name`, `description`, `label`, etc given as `foreman_organization`. See example below:
```
foreman_server_url: url
foreman_username: user
foreman_password: pass
foreman_validate_certs: no
foreman_organizations: 
  - name: raleigh
    label: rdu
    state: present
  - name: lanai 
    description: pacific datacenter 
  - name: default
    label: boring
    state: absent
```

the first set of authentication vars will go under the "Variables" section of your README and the `foreman_organizations` structure gets documented under the "Organization Data Structure" section of your README.

- [x] update loop vars
- [x] update README
- [x] update examples

**Comment by willtome** [comment]
Closes #1109

File: roles/organizations/README.md [hunk 1]
```diff
@@ -1,23 +1,20 @@
 # foreman.organizations
```
Comment (willtome) [inline roles/organizations/README.md]: update to fully qualified collection name `theforeman.foreman.organizations`

File: roles/organizations/README.md [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
 |Variable Name|Default Value|Required|Description|Example|
 |:---:|:---:|:---:|:---:|:---:|
-|`tower_state`|"present"|no|The state all objects will take unless overridden by object default|'absent'|
-|`tower_hostname`|""|yes|URL to the Ansible Tower Server.|127.0.0.1|
-|`validate_certs`|`False`|no|Whether or not to validate the Ansible Tower Server's SSL certificate.||
-|`tower_username`|""|yes|Admin User on the Ansible Tower Server.||
-|`tower_password`|""|yes|Tower Admin User's password on the Ansible Tower Server.  This should be stored in an Ansible Vault at vars/tower-secrets.yml or elsewhere and called from a parent playbook.||
-|`tower_oauthtoken`|""|yes|Tower Admin User's token on the Ansible Tower Server.  This should be stored in an Ansible Vault at or elsewhere and called from a parent playbook.||
-|`tower_organizations`|`see below`|yes|Data structure describing your organzation or organizations Described below.||
+|`foreman_server_url`|"present"|no|URL of the Foreman server
```
Comment (willtome) [inline roles/organizations/README.md]: `present` is not a valid default for `foreman_server_url`

File: roles/organizations/README.md [hunk 3]
```diff
@@ -1,23 +1,20 @@
...
+|`foreman_username`|""|yes|Username accessing the Foreman Server.||
```
Comment (willtome) [inline roles/organizations/README.md]: username and password should not be required. check code for `default(omit)` to verify

File: roles/organizations/README.md [hunk 4]
```diff
@@ -1,23 +1,20 @@
...
+|`foreman_password`|""|yes|Password of user accessing the Foreman Server.  This should be stored at vars/foreman-secrets.yml or elsewhere 
+|`validate_certs` |""|no|Whether or not to verify the TLS certificates of the Foreman server.
```
Comment (willtome) [inline roles/organizations/README.md]: use `foreman_` prefix for validate_certs variable name. document default value

File: roles/organizations/README.md [hunk 5]
```diff
@@ -1,23 +1,20 @@
...
+|`foreman_description` |""|no|Description of named organization
```
Comment (willtome) [inline roles/organizations/README.md]: `foreman_description` doesn't make sense here. perhaps you meant `foreman_organizations` as shown in your example?

File: roles/organizations/README.md [hunk 6]
```diff
@@ -33,52 +30,9 @@ This role accepts two data models. A simple straightforward easy to maintain mod
...
-  description: Default
-  max_hosts: 0
-  custom_virtualenv:
-  related:
-    notification_templates_started: []
-    notification_templates_success: []
-    notification_templates_error: []
-    notification_templates_approvals: []
-```
+|`foreman_description` |""|no|Description of Named Organization|
```
Comment (willtome) [inline roles/organizations/README.md]: no `foreman_` prefix required

File: roles/organizations/README.md [hunk 7]
```diff
@@ -33,52 +30,9 @@ This role accepts two data models. A simple straightforward easy to maintain mod
...
+|`label`|`False`|no|Custom Label|
```
Comment (willtome) [inline roles/organizations/README.md]: False is not valid default for label

**Review by willtome — CHANGES_REQUESTED** [review]

File: roles/organizations/README.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+An Ansible Role to create Organizations in Foreman.
+
+## Variables
+|Variable Name|Default Value|Required|Description|Example|
+|:---:|:---:|:---:|:---:|:---:|
+|`foreman_server_url`|""|no|URL of the Foreman server.|
+|`foreman_username`|""|no|Username accessing the Foreman Server.|
+|`foreman_password`|""|no|Password of user accessing the Foreman Server.  This should be stored at vars/foreman-secrets.yml or elsewhere.|
+|`foreman_validate_certs` |"foreman_"|no|Whether or not to verify the TLS certificates of the Foreman server.|
+|`foreman_organizations`|"present"|no|Manage the Foreman Organization.|
```
Comment (willtome) [inline roles/organizations/README.md]: Default would be `[]`

File: roles/organizations/README.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+## Description
+An Ansible Role to create Organizations in Foreman.
+
+## Variables
+|Variable Name|Default Value|Required|Description|Example|
+|:---:|:---:|:---:|:---:|:---:|
+|`foreman_server_url`|""|no|URL of the Foreman server.|
+|`foreman_username`|""|no|Username accessing the Foreman Server.|
+|`foreman_password`|""|no|Password of user accessing the Foreman Server.  This should be stored at vars/foreman-secrets.yml or elsewhere.|
+|`foreman_validate_certs` |"foreman_"|no|Whether or not to verify the TLS certificates of the Foreman server.|
```
Comment (willtome) [inline roles/organizations/README.md]: Default would be `""`

**Review by willtome — CHANGES_REQUESTED** [review]

**Review by willtome — APPROVED** [review]

**Comment by ehelms** [comment]
This diverges from our current variable naming conventions and therefore I believe it should be blocked on either updating to our current standard or waiting until we resolve the proposed naming changes so that we have consistency across our roles. 

The issue under consideration: https://github.com/theforeman/foreman-ansible-modules/issues/1118

**Comment by willtome** [comment]
Opened #1131 to meet the current naming scheme. Leaving the PR as is until #1118 is resolved and this can be merged or closed.

**Comment by evgeni** [comment]
Given we ended up with having a prefix, this is the way to go :)

**Comment by evgeni** [comment]
taken over in #1204

**Pull request closed without merging**